### PR TITLE
Remove `name` field from `Comment`

### LINKED ISSUE #46: Remove name from Comment
The name in Comment isn't used, so we should probably remove it, as discussed here
https://github.com/exercism/elm-analyzer/pull/44#discussion_r1006526840

> Actually to be honest, "maxPower doesn't use max" is not used anywhere, it's only there for the sake of humans explaining when this comment would pop up.
It makes tests and rules a little bit more readable, that's all. Maybe it's not worth it...

name property defined here:
https://github.com/exercism/elm-analyzer/blob/dfe46ca127fe8fe15da508ec6216036e42e4ee62/src/Comment.elm#L33

### PR DESCRIPTION
**Pull request by jiegillet**
Closes #46 

Full disclosure: this started as an exploration of using `elm-review` with `--extract`, because that's what I wanted to use for the tags analysis.

[Data extraction](https://jfmengels.net/elm-review-insights/) is a new-ish feature of elm-review that allow a rule to export arbitrary JSON data instead of a `Review.Rule.Error`. Since our analysis process is complicated and involves stuffing a JSON string into an error message and decoding it afterwards, I thought I could use data extraction instead to save some time.

It turns out that  `Analyzer.functionCall` was quite simple to port to a data extractor, however there were some issues:
- We are using `Rule.filterErrorsForFiles` to only run rules on specific file names corresponding to their exercise, however it seems that it's not working with data extractors. It's of course possible to do, but would be more verbose because I would need to pass the file name in each call of `Analyzer.functionCall` instead of specifying the file name once per module.
- Not all exercise rules are using `Analyzer.functionCall`, instead they define their own rule, we would need to change them too, and using data extractors requires quite a bit more boilerplate.
- I'm not 100% sure because I didn't get to the testing part, but I think refactoring tests would have involved meaningful changes.
- We would still need to produce normal errors and then decode them again anyway because that's how common rules work, no way around that.

So in the end, I decided that it's not worth doing at this time. I still think it's the right call for the tags though, because they would probably use a different process than `Analyzer.functionCall`.

Sometime during these considerations, I realized that we were not using the slug fields for anything, and so I removed them, and then I thought it was time to remove the unused named field as well. Whenever some tests fail, as long as the test name is meaningful ("that function doesn't do what it's supposed to" kind of name) then there is no confusion at all, even if several rules can lead to the same comment sometimes.

### CODE REVIEW

**Review by ceddlyburge — APPROVED** [review]

**Comment by jiegillet** [comment]
@exercism/maintainers-admin Can I get an approval please? 💛

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**
